### ISSUE TITLE
How to collect data for data.json

### ISSUE THREAD

**Issue by GunGunGun** [TinkleWorld/fkg-hs-player#1]
Hi, thanks very much for this awesome project, I want to update data.json to add new characters to FKG HS Player, but I have no idea what is the DMM API URL to get those IDs, can you let me know what is the URL you use to get those IDs or do I need to reverse engineering the game ? Thanks!

**Comment by Amarillys** [comment]
Thanks for your reply. It's late now. Please wait for me. I will tell you tomorrow. :)

**Comment by Amarillys** [comment]
Hi, I have updated the repo, I think it could help you.
c1c72e167d9fda3a96021f565624948dffc616a2

**Comment by GunGunGun** [comment]
> Hi, I have updated the repo, I think it could help you. [c1c72e1](https://github.com/TinkleWorld/fkg-hs-player/commit/c1c72e167d9fda3a96021f565624948dffc616a2)

Thanks a lot and sorry for late replying, I also found another source to get game data, it's https://flowerknight.fandom.com/wiki/Module:MasterCharacterData?action=edit

I parsed the data into JSON and it worked pretty well and the data is very up-to-date, I'll try to use your latest commit later, thanks a lot!

**Comment by Amarillys** [comment]
> > Hi, I have updated the repo, I think it could help you. [c1c72e1](https://github.com/TinkleWorld/fkg-hs-player/commit/c1c72e167d9fda3a96021f565624948dffc616a2)
> 
> Thanks a lot and sorry for late replying, I also found another source to get game data, it's https://flowerknight.fandom.com/wiki/Module:MasterCharacterData?action=edit
> 
> I parsed the data into JSON and it worked pretty well and the data is very up-to-date, I'll try to use your latest commit later, thanks a lot!

Hey, It looks like a very convient data provider. :)
It may help me updating date next time.

**Comment by GunGunGun** [comment]
> Hey, It looks like a very convient data provider. :) It may help me updating date next time.

Yeah, good thing about above link is that they also provide English Name, Nation, many many useful datas of Flower Girls, so it's always good to use English Name just for reference and convenience.

**Comment by SXK485** [comment]
> Hi, I have updated the repo, I think it could help you. [c1c72e1](https://github.com/TinkleWorld/fkg-hs-player/commit/c1c72e167d9fda3a96021f565624948dffc616a2)

Thank you for the update. I tried using the code from operation.txt to get charaData but failed. I can't access the indexedDB data sourced from https://dugrqaqinbtcq.cloudfront.net/ because of the same-origin policy restrictions. How did you solve this problem?

**Comment by Amarillys** [comment]
> > Hi, I have updated the repo, I think it could help you. [c1c72e1](https://github.com/TinkleWorld/fkg-hs-player/commit/c1c72e167d9fda3a96021f565624948dffc616a2)
> 
> Thank you for the update. I tried using the code from operation.txt to get charaData but failed. I can't access the indexedDB data sourced from https://dugrqaqinbtcq.cloudfront.net/ because of the same-origin policy restrictions. How did you solve this problem?

You can take a look at you devtools's domain. Its default value is top(dmm.co.jp). Maybe you could switch it to https://dugrqaqinbtcq.cloudfront.net/  and run the code under it.

**Comment by SXK485** [comment]
> > > Hi, I have updated the repo, I think it could help you. [c1c72e1](https://github.com/TinkleWorld/fkg-hs-player/commit/c1c72e167d9fda3a96021f565624948dffc616a2)
> > 
> > 
> > Thank you for the update. I tried using the code from operation.txt to get charaData but failed. I can't access the indexedDB data sourced from https://dugrqaqinbtcq.cloudfront.net/ because of the same-origin policy restrictions. How did you solve this problem?
> 
> You can take a look at you devtools's domain. Its default value is top(dmm.co.jp). Maybe you could switch it to https://dugrqaqinbtcq.cloudfront.net/ and run the code under it.

Hey, your tip worked! I've learned a new way to get data.json. Thanks for your help.
